src/NServiceBus.Core/Performance/MessageDurability/DetermineMessageDurabilityBehavior.cs

```diff
@@ -22,7 +21,7 @@ public override Task Invoke(OutgoingLogicalMessageContext context, Func<Task> ne
             {
                 context.AddDeliveryConstraint(new NonDurableDelivery());
 
-                context.SetHeader(Headers.NonDurableMessage,true.ToString());
+                context.Headers[Headers.NonDurableMessage] = true.ToString();
```

Comment:
I asked this once long ago on StackOverflow: it will deserialize bools with those values, yes. And AFAIK, it does an `OrdinalIgnoreCase` compare when doing that, so the casing doesn't *really* matter, but it's nice to be consistent.